src/gui/qgssourceselectprovider.h

```diff
@@ -0,0 +1,71 @@
+/***************************************************************************
+  qgssourceselectprovider.h - QgsSourceSelectProvider
+
+ ---------------------
+ begin                : 1.9.2017
+ copyright            : (C) 2017 by Alessandro Pasotti
+ email                : apasotti at boundlessgeo dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSSOURCESELECTPROVIDER_H
+#define QGSSOURCESELECTPROVIDER_H
+
+
+#include "qgis_gui.h"
+#include "qgis.h"
+#include "qgsguiutils.h"
+#include "qgsproviderregistry.h"
+#include "qgsabstractdatasourcewidget.h"
+
+class QString;
+class QWidget;
+
+/** \ingroup gui
+ * This is the interface for those who want to add entries to the \see QgsDataSourceManagerDialog
```

Comment:
(totally offtopic to this PR) - I don't think the \see is necessary here. My understanding is that doxygen will auto detect the class name and link it automatically. I'd love to know for certain either way though.

Reply:
Inline things should be tagged with `\link  \endlink` I think, `\see` is for a list of related topics at the end of the block IIRC.

Reply:
Ok - just to confirm. Doxygen DOES auto link class names. See https://qgis.org/api/classQgsPropertyOverrideButton.html#details . The links to QgsProperty/etc were auto inserted.